test/result/function/func_json_extract.result

```diff
@@ -34,6 +34,21 @@ json_extract({"a":[1,2,3,{"a":4}]}, $**.a)
 select json_extract('{"a":[1,2,3,{"a":4}]}','$.a[*].a');
 json_extract({"a":[1,2,3,{"a":4}]}, $.a[*].a)
 4
+select json_extract('{"a":1}','$[0]');
```

Comment:
Isn't the result suppose to be null?  $[0] is the first element of an array, correct?     This should be $, that return the whole json.

And don't you need to quote the json?

Reply:
here just to keep same behavior with mysql.
In mysql, if the doc is object and current path if [0] it will continue to parse current object doc.
But if index>0, such as [1] for object doc, return null.

Reply:
where need to quote?